.github/workflows/workflow.yml

```
@@ -71,8 +71,31 @@ jobs:
           CI: true
           TESTOPTS: -v --no-show-detail-immediately
 
+  pure_ruby:
+    name: >-
+      pure ruby (${{ matrix.os }} ${{ matrix.ruby }})
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 10
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-20.04, ubuntu-22.04, ubuntu-24.04 ]
+        ruby: [ '2.7', '3.0', '3.1', '3.2', '3.3', head ]
```

@sodabrew What do you think about only running pure ruby CI on non-EOL ruby versions?
```suggestion
        ruby: [ '3.1', '3.2', '3.3', head ]
```

---

Oh 2.7 is deprecated already, is it 😭